[docs][ENG-4933] add information about CocoaPods cache server (#19050)
expo/expo: docs/pages/build-reference/infrastructure.md

- Android workers run on Kubernetes in an isolated environment
  - Every build gets its own container running on a dedicated Kubernetes node
  - Build resources: 4 CPU, 12 GB RAM

> - npm cache deployed with Kubernetes. [Learn more](caching/#javascript-dependencies)

Reviewer: ```suggestion
- [npm cache deployed with Kubernetes](caching/#javascript-dependencies)
```

I think we can provide the link directly to the subject rather than adding a "Learn more".

> - Maven cache deployed with Kubernetes. [Learn more](caching/#android-dependencies)

Reviewer: ```suggestion
- [Maven cache deployed with Kubernetes](caching/#android-dependencies)
```

  - Every build gets its own fresh macOS VM
  - Hardware: Intel(R) Core(TM) i7-8700B CPU (6 cores/12 threads), 64 GB RAM
  - Build resource limits: 3 cores, 12 GB RAM

> - NPM cache. [Learn more](caching/#javascript-dependencies)

Reviewer 2: https://css-tricks.com/start-sentence-npm/#:~:text=Asking%20this%20question%20was%20a,precedent%20for%20not%20capitalizing%20it.

Author: We used capitalized "npm" in line 33 and I thought it was correct 😄 I will change it as well

> - CocoaPods cache. [Learn more](caching/#ios-dependencies)

Reviewer 2: Maybe similarly to npmrc it would be useful to show what the Podfile will look like.

- CocoaPods cache. [Learn more](caching/#ios-dependencies)
- `~/.npmrc`

  ```ini

>   registry=http://10.254.24.8:4873

Reviewer 2: Have we changed that?

Author: This was staging URL, I changed it to production

  - Hardware: Intel(R) Core(TM) i7-8700B CPU (6 cores/12 threads), 64 GB RAM
  - Build resource limits: 3 cores, 12 GB RAM

> - npm cache. [Learn more](caching/#javascript-dependencies)
> - CocoaPods cache. [Learn more](caching/#ios-dependencies)

Reviewer: ```suggestion
- [npm cache](caching/#javascript-dependencies)
- [CocoaPods cache](caching/#ios-dependencies)
```

- `~/.npmrc`

  ```ini
  registry=http://10.254.24.9:4873
  ```

- `~/.yarnrc.yml`

  ```yml
  unsafeHttpWhitelist:
    - '*'
  npmRegistryServer: 'registry=http://10.254.24.9:4873'
  enableImmutableInstalls: false
  ```

> - cocoapods-nexus-plugin. [Learn more](https://github.com/expo/eas-build/tree/main/packages/cocoapods-nexus-plugin)

Reviewer: ```suggestion
- [cocoapods-nexus-plugin](https://github.com/expo/eas-build/tree/main/packages/cocoapods-nexus-plugin)
```

  - Hardware: Intel(R) Core(TM) i7-8700B CPU (6 cores/12 threads), 64 GB RAM
  - Build resource limits: 3 cores, 12 GB RAM
- npm cache. [Learn more](caching/#javascript-dependencies)
- CocoaPods cache. [Learn more](caching/#ios-dependencies)
- `~/.npmrc`

  ```ini
  registry=http://10.254.24.9:4873
  ```

- `~/.yarnrc.yml`

  ```yml
  unsafeHttpWhitelist:
    - '*'
  npmRegistryServer: 'registry=http://10.254.24.9:4873'
  enableImmutableInstalls: false
  ```

> - Project's `Podfile` will have CocoaPods cache server URL added as a source during the build step called `Install pods` (if `Podfile` already contains explicitly specified `https://cdn.cocoapods.org/` source, it will be replaced with CocoaPods cache server URL).

Reviewer 2: ```suggestion
- For iOS builds, EAS Build updates your `Podfile` during the `Install pods` build phase. We set the CocoaPods cache server URL as the `source`. If your `Podfile` includes `source "https://cdn.cocoapods.org/"`, it will be replaced with the CocoaPods cache server URL.
```
